### PR TITLE
When attempting to write str with single quote suggest double quotes

### LINKED ISSUE #26101: Poor error when using single quotes with `println!`
```
fn main() {
    println!('{}', 5);
}
```

This error leaves much to be desired:

```
hello.rs:2:14: 2:16 error: unterminated character constant: '{
hello.rs:2     println!('{}', 5);
                        ^~
```

The issue is that you need double quotes. 

originally found in http://joelmccracken.github.io/entries/a-simple-web-app-in-rust-pt-2b/ /cc @joelmccracken

### PR DESCRIPTION
**Pull request by estebank**
Fix #26101.

### CODE REVIEW

**Comment by estebank** [comment]
New output:

**Comment by leonardo-m** [comment]
Multi-character character constant (MCCC) as in C are not supported yet in Rust, so this error message makes sense. To define MCCC we can use a macro.

File: src/test/ui/suggestions/str-as-char.stderr [hunk 1]
```diff
@@ -0,0 +1,12 @@
+error: character literal may only contain one codepoint
+  --> $DIR/str-as-char.rs:12:14
+   |
+12 |     println!('{}', 5);
+   |              ^^^^
+help: if you meant to write a `str` literal use double quotes instead
+   |
+12 |     println!("{", 5);
```
Comment (bjorn3) [inline src/test/ui/suggestions/str-as-char.stderr]: Where did the `}` go?

File: src/test/ui/suggestions/str-as-char.stderr [hunk 2]
```diff
@@ -0,0 +1,12 @@
+error: character literal may only contain one codepoint
+  --> $DIR/str-as-char.rs:12:14
+   |
+12 |     println!('●●');
+   |              ^^^^
+help: if you meant to write a `str` literal use double quotes instead
```
Comment (zackmdavis) [inline src/test/ui/suggestions/str-as-char.stderr]:
```
if you meant to write a `str` literal, use double quotes
```
(comma separating conditional clause and consequent)

**Comment by petrochenkov** [comment]
How I see the best solution - single-quoted strings are actually lexed and recovered as strings sharing common code with double-quoted strings.
@bors r+

**Comment by estebank** [comment]
@bors r=petrochenkov

rebased to latest master, no actual merge conflict found

**Pull request merged**